src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -259,15 +277,60 @@ impl ScalarFunc {
         }
         self.children[i as usize].eval_string(ctx, row)
     }
+
+    #[inline]
+    pub fn trim_1_arg<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let s = try_opt!(self.children[0].eval_string_and_decode(ctx, row));
+        let r = s.trim_matches(SPACE_CHAR);
+        Ok(Some(Cow::Owned(r.to_string().into_bytes())))
+    }
+
+    #[inline]
+    pub fn trim_2_args<'a, 'b: 'a>(
```

Comment:
Can we make `trim_2_args` to call `trim_1_arg`, `trim_3_args` to call `trim_2_args` to reuse codes?

Reply:
hi, I had extract a trim helper function, based on the trim_3_args, which is the most generic of the three.